lib/synapse/haproxy.rb

```diff
@@ -714,6 +714,11 @@ def generate_backend_stanza(watcher, config)
             log.info "synapse: restart required because haproxy_server_options changed for #{backend_name}"
             @restart_required = true
           end
+          if (old_backend.fetch('weight', "") !=
```

Comment:
We detect a restart as long as the output config file is different: https://github.com/airbnb/synapse/blob/master/lib/synapse/haproxy.rb#L844. So you're right to point out this block of code as being out-of-place. I'm writing some tests to see if we even need any of the logic here.